lib/services/project-templates-service.ts

```diff
@@ -6,31 +6,32 @@ temp.track();
 export class ProjectTemplatesService implements IProjectTemplatesService {
 
 	public constructor(private $errors: IErrors,
-						private $fs: IFileSystem,
-						private $logger: ILogger,
-						private $npm: INodePackageManager,
-						private $npmInstallationManager: INpmInstallationManager) { }
+		private $fs: IFileSystem,
+		private $logger: ILogger,
+		private $npm: INodePackageManager,
+		private $npmInstallationManager: INpmInstallationManager,
+		private $analyticsService: IAnalyticsService) { }
 
 	public prepareTemplate(originalTemplateName: string, projectDir: string): IFuture<string> {
 		return ((): string => {
-			let realTemplatePath: string;
-			if(originalTemplateName) {
+			let templateName = originalTemplateName || constants.RESERVED_TEMPLATE_NAMES["default"],
+				version: string = null;
+
+			if (originalTemplateName) {
 				// support <reserved_name>@<version> syntax
 				let data = originalTemplateName.split("@"),
-					name = data[0],
-					version = data[1];
-
-				if(constants.RESERVED_TEMPLATE_NAMES[name.toLowerCase()]) {
-					realTemplatePath = this.prepareNativeScriptTemplate(constants.RESERVED_TEMPLATE_NAMES[name.toLowerCase()], version, projectDir).wait();
-				} else {
-					// Use the original template name, specified by user as it may be case-sensitive.
-					realTemplatePath = this.prepareNativeScriptTemplate(name, version, projectDir).wait();
-				}
-			} else {
-				realTemplatePath = this.prepareNativeScriptTemplate(constants.RESERVED_TEMPLATE_NAMES["default"], null/*version*/, projectDir).wait();
+					name = data[0];
+
+				version = data[1];
+
+				templateName = constants.RESERVED_TEMPLATE_NAMES[name.toLowerCase()] || name;
```

Comment:
In fact the case is handled - when `originalTemplateName` is `app`, the `name` variable will be set to `app`.
At this point `templateName` will be set to `app`.

At another point, when user passes `--template App`, `originalTemplateName` will be set to `App`, the `name` variable will be set to `App` and the `templateName` will be set to `App` again.